core/data_stream.py

```diff
@@ -236,12 +236,26 @@ def write_symbol(self, s):
 
 
 class Uint8FileDataStream(FileDataStream):
-    """reads Uint8 numbers written to a file
```

Comment:
Please add a short test ensuring that this is lossless

Reply:
will do, thanks

Reply:
done in revision